plugins/kubernetes/app/models/kubernetes/template_filler.rb

```diff
@@ -44,7 +44,11 @@ def to_hash(verification: false)
           make_stateful_set_match_service if kind == 'StatefulSet'
           set_pre_stop if kind == 'Deployment'
           set_name
-          (set_replica_target || validate_replica_target_is_supported) if kind != 'PodTemplate'
+          if ['Deployment', 'StatefulSet'].include?(kind)
```

Comment:
this is a change in behavior, correct? Previously it would attempt to set replicas on anything (comment "custom resource that has replicas set on itself or it's template")

Reply:
yeah, that was 1-off logic for shardeddeploments that we no longer need 🤞

Reply:
ah I see!